src/store/index.ts

```diff
@@ -0,0 +1,664 @@
+import { StoreOptions } from 'vuex'
```

Comment:
**Suggestion**: Can I ask you to undo the renaming of this file, please? It’s not necessary and I don’t want to promote the possibility of importing from a path for a directory with an index file. Importing from `$PATH/storeConfig` is far more clear than importing from `$PATH` and is also much less ambiguous for module resolution algorithms which have to run more checks for resolving paths when importing using that style.

Reply:
I'm pretty sure it is necessary for all the contents not to be `storeConfig` otherwise we have circular dependency issues (we access the container in the `storeConfig` file so we can't then import that same file from the container file). This is just down to history/baggage, i.e. most of the app currently accesses the store from a hardcoded import instead of an injectable dependency. I don't want to go through the codebase right now changing all this, so I did the smallest amount of churn possible to keep the same functionality throughout the app, but make the app always access the store via the service container.

This was all because we need to make KumaApi injectable (the store uses KumaApi a great deal, but now we need to inject that in)

Happy to rename it to anything you like, let me know what works for you. I just chose `index` as its an _index_ of everything to do with the `store` (apart from the actual instances, which usually would come straight from the service container) and I couldn't think of anything else off the top of my head. It also makes things quite nice to work with imo as you can think of `index` exposing a public interface, and the rest should be private to the module - kinda like a JS module.

Reply:
> I'm pretty sure it is necessary for all the contents not to be storeConfig otherwise we have circular dependency issues (we access the container in the storeConfig file so we can't then import that same file from the container file).

Can you please explain what this means? Where does a circular dependency happen? Why does this prevent you from calling this file anything but index.ts?

> Happy to rename it to anything you like, let me know what works for you.

storeConfig.ts is perfectly adequate.

> I just chose index as its an index of everything to do with the store

The current file name of storeConfig does a better job at indicating what it contains because there is likely only going to be one of it in the entire code base whereas with the `$PATH/index.ts` pattern, there are potentially dozens of index files. It also allows developers to quickly open this file using an editor’s command palette by typing `storeConfig` (or some form of abbreviation of it). That’s also not possible if everything is in an index.ts file.

Reply:
> Why does this prevent you from calling this file anything but index.ts?

It doesn't prevent me from calling it anything but `index.ts` it prevents me from putting all this in `storeConfig.ts`, anything else is fine.

> storeConfig.ts is perfectly adequate.

it prevents me from putting all this in `storeConfig.ts`, anything else is fine.

> Where does a circular dependency happen? 

If you look in storeConfig as its stands in this PR on the first line:

```
import { get, TOKENS } from '@/services'
```

If the things I need to import from inside the '@/services' file (i.e. `production` or `development`) import the the `@/services` file itself (as is the case with `storeConfig`) then there is a circular dependency.

Reply:
Can you think of a way to _not_ rename this file? I really don’t want to lose the ability to look up implementation files like this by popping open the command palette and typing `storeConfig`.

Reply:
You don't lose anything, you just type `store/index` instead of `storeConfig`, or am I not following here? I just changed the name of the file?

I can change it to `storeConfig` if you like, but then I have to change the name of the current `storeConfig` to something else. I don't mind, just let me know what fits you if you don't want `index`, just we need 2 files not 1.

Reply:
Ok so as discussed offline, I removed the one place where we were previously importing `storeConfig` (on `master` we import it from 2 places, but one of those was removed in this PR in favour of using the service container instead). This means we don't need to import the container into `storeConfig.ts` which then means I can then use `storeConfig.ts` for the contents of `index` without running into circular dependency issues, and lastly `index` went in the bin 🎉 

I think thats all that sorted now then?

Reply:
Yep, that’s all good now. 👍🏻